### ISSUE TITLE
VovNet - Model Card

### ISSUE THREAD

**Issue by Sudharsan-V** [tenstorrent/tt-metal#17934]
Torch Reference code: [https://github.com/huggingface/pytorch-image-models/blob/main/timm/models/vovnet.py](https://github.com/huggingface/pytorch-image-models/blob/main/timm/models/vovnet.py)

Graph:  [Vovnet_graph.gv.pdf](https://github.com/user-attachments/files/18842249/Vovnet_graph.gv.pdf)

Model architecture: [Architecture.docx](https://docs.google.com/document/d/1--5y38zxc84zXfSSjkXZxzdbrEDVXbqpVLJCP2wu3-4/edit?usp=sharing)

Model Architecture sheet: [sheet](https://docs.google.com/spreadsheets/d/1ehXndNVa3m1x8jCkuSHxa-P41q-hrZME4AWJaZLGrPU/edit?usp=sharing)

Modules:
```
Stem
      ConvNormAct
      SeperableConvNormAct

Stages
       OsaStage
            Osablock
                 Convreduction
                      Conv_mid
                          Conv_concat
                               attn

head
  
```

Ops:

- Conv2d
- Batchnorm
- relu
- Identity
- Harshsigmoid
- linear
- SelectAdaptivepool2d

**Comment by Sudharsan-V** [comment]
The fused conv op works correctly with random weights, but it fails when using VoVNet model weights.
corresponding issue [#18137](https://github.com/tenstorrent/tt-metal/issues/18137#issuecomment-2678262003)

**Comment by Sudharsan-V** [comment]
The end-to-end pipeline for the VoVNet model has been enabled. However, the model does not perform as expected with real weights due to issues with fused conv (https://github.com/tenstorrent/tt-metal/issues/18137#issuecomment-2678262003). With random weights, the model achieves a PCC > 0.99.

As there is an issue with fused convolution, it has been replaced with TTNN conv and TTNN batch norm. With this setup, the model successfully achieves a PCC > 0.99 with random weight.

corresponding [commit](https://github.com/tenstorrent/tt-metal/compare/main...sudharsan/ttnn_vovnet)

**Comment by Sudharsan-V** [comment]
@saichandax , The issue is closed but the problem still persists.

**Comment by saichandax** [comment]
@Sudharsan-V , please tag the relevant issue that is blocking this model.

**Comment by Sudharsan-V** [comment]
The end-to-end pipeline for the VoVNet model has been completed. Currently, the model achieves a PCC ~ 0.7. Individual sub-modules have been tested and are passing with a PCC greater than 0.99, as expected.

The overall drop in PCC may be due to the sequential use of the osa_stage sub-module (which itself has PCC > 0.99 in unit tests). Since this sub-module appears four times in sequence, small errors may have accumulated, leading to the reduced final PCC.

As part of debugging, Replacing ttnn.global_avg_pool2d to nn.AdaptiveAvgPool2d in [classifier_head.py](https://github.com/tenstorrent/tt-metal/blob/sudharsan/ttnn_vovnet/models/experimental/functional_vovnet/tt/classifier_head.py) improved the pcc from 0.7 to 0.92

Corresponding issue [#23474](https://github.com/tenstorrent/tt-metal/issues/23474)
